src/sql/workbench/test/common/editorReplacerContribution.test.ts

```diff
@@ -0,0 +1,253 @@
+		const editorService = new MockEditorService();
+		const instantiationService = workbenchInstantiationService();
+		instantiationService.stub(IEditorService, editorService);
+		const contrib = instantiationService.createInstance(EditorReplacementContribution);
+		const input = instantiationService.createInstance(FileEditorInput, URI.file('/test/file.iynb'), undefined, 'notebook');
+		const response = editorService.fireOpenEditor(input, undefined, undefined as IEditorGroup);
+		assert(response?.override);
+		const newinput = <any>(await response.override) as EditorInput; // our test service returns this so we are fine to cast this
+
+		assert(newinput instanceof NotebookInput);
+
+		contrib.dispose();
+	});
+
+	test('does replace notebook file input using input mode', async () => {
+		const editorService = new MockEditorService();
+		const instantiationService = workbenchInstantiationService();
+		instantiationService.stub(IEditorService, editorService);
+		const contrib = instantiationService.createInstance(EditorReplacementContribution);
+		const input = instantiationService.createInstance(UntitledTextEditorInput, URI.file('/test/file'), false, undefined, undefined, undefined);
+		const response = editorService.fireOpenEditor(input, undefined, undefined as IEditorGroup);
+		assert(response?.override);
+		const newinput = <any>(await response.override) as EditorInput; // our test service returns this so we are fine to cast this
+
+		assert(newinput instanceof QueryEditorInput);
+
+		contrib.dispose();
+	});
+
+	test('does not replace editors that it shouldnt', async () => {
+		const editorService = new MockEditorService();
+		const instantiationService = workbenchInstantiationService();
+		instantiationService.stub(IEditorService, editorService);
+		const contrib = instantiationService.createInstance(EditorReplacementContribution);
+		const untitled = instantiationService.createInstance(UntitledTextEditorInput, URI.file('/test/file'), false, undefined, undefined, undefined);
+		const input = instantiationService.createInstance(UntitledQueryEditorInput, '', untitled, undefined);
+		const response = editorService.fireOpenEditor(input, undefined, undefined as IEditorGroup);
+		assert(response === undefined);
+
+		contrib.dispose();
+	});
+
+	test('does not replace editors if it doesnt have a replacer', async () => {
+		const editorService = new MockEditorService();
+		const instantiationService = workbenchInstantiationService();
+		instantiationService.stub(IEditorService, editorService);
+		const contrib = instantiationService.createInstance(EditorReplacementContribution);
+		const input = instantiationService.createInstance(UntitledTextEditorInput, URI.file('/test/file.unknown'), false, undefined, undefined, undefined);
+		const response = editorService.fireOpenEditor(input, undefined, undefined as IEditorGroup);
+		assert(response === undefined);
+
+		contrib.dispose();
+	});
+});
+
+class MockEditorService extends TestEditorService {
```

Comment:
Move these test classes into their own files? The TestModeService for sure seems like that should be separate since that's used in a lot of places